cmd/prchecker/main_test.go

```diff
@@ -0,0 +1,232 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-github/github"
+)
+
+func TestStringAllowsModuleSpan(t *testing.T) {
+	var tests = []struct {
+		body          string
+		matchExpected bool
+	}{
+		{"foo", false},
+		{"don't ALLOW_MODULE_SPAN", false},
+		{"ALLOW_MODULE_SPAN", true},       // PR body is just exception
+		{"> ALLOW_MODULE_SPAN", true},     // support quoting exception
+		{"\t ALLOW_MODULE_SPAN\t ", true}, // support whitespace padding
+		{"ALLOW_MODULE_SPAN\nat start of file", true},
+		{"at end of file\nALLOW_MODULE_SPAN", true},
+	}
+
+	for _, tt := range tests {
+		testname := fmt.Sprintf("does %v allow spanning %v", tt.body, tt.matchExpected)
+
+		t.Run(testname, func(t *testing.T) {
+			result, err := StringAllowsModuleSpan(tt.body)
+
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			if result != tt.matchExpected {
+				t.Errorf("got %t, want %t", result, tt.matchExpected)
+			}
+		})
+	}
+}
+
+func TestIsModuleSpanAllowed(t *testing.T) {
+	var tests = []struct {
+		body          string
+		matchExpected bool
+	}{
+		{"don't ALLOW_MODULE_SPAN", false},
+		{"ALLOW_MODULE_SPAN", true},
+	}
+
+	for _, tt := range tests {
+		testname := fmt.Sprintf("does %v allow spanning %v", tt.body, tt.matchExpected)
+
+		t.Run(testname, func(t *testing.T) {
+
+			client := FakeGitHubService{
+				pullRequest: &github.PullRequest{
+					Body: &tt.body,
+				},
+			}
+
+			result, _ := IsModuleSpanAllowed(&client, 1)
+
+			if result != tt.matchExpected {
+				t.Errorf("got %t, want %t", result, tt.matchExpected)
+			}
+		})
+	}
+}
+
+func TestGetCommitChanges(t *testing.T) {
+	testname := fmt.Sprintf("validate get commit changes converts to changeset")
+
+	t.Run(testname, func(t *testing.T) {
```

Comment:
This is not needed if there is only one testcase.

Reply:
In this case would the test name be the function's signature?

Reply:
Yeah. Subtests are only needed when we have multiple test cases in one test function.